docs/word/word-add-ins-events.md

```diff
@@ -0,0 +1,199 @@
+
+> [!CAUTION]
+> When an object to which events are registered is deleted (e.g., a content control with an `onDataChanged` event registered), the event handler no longer triggers but remains in memory until the add-in or Word session refreshes or closes.
+
+### Events and coauthoring
+
+With coauthoring, multiple people can work together and edit the same Word document simultaneously. For events that can be triggered by a coauthor, such as `onParagraphChanged`, the corresponding **Event** object will contain a **source** property that indicates whether the event was triggered locally by the current user (`event.source = Local`) or was triggered by the remote coauthor (`event.source = Remote`).
+
+## Register an event handler
+
+The following code sample registers an event handler for the `onParagraphChanged` event in the document. The code specifies that when content changes in the document, the `handleChange` function should run.
+
+```js
+await Word.run(async (context) => {
+    eventContext = context.document.onParagraphChanged.add(handleChange);
+
+    await context.sync();
+    console.log("Event handler successfully registered for onParagraphChanged event in the document.");
+}).catch(errorHandlerFunction);
+```
+
+## Handle an event
+
+As shown in the previous example, when you register an event handler, you indicate the function that should run when the specified event occurs. You can design that function to perform whatever actions your scenario requires. The following code sample shows an event handler function that simply writes information about the event to the console.
+
+```js
+async function handleChange(event) {
+    await Word.run(async (context) => {
+        await context.sync();        
+        console.log("Type of event: " + event.type);
+        console.log("Source of event: " + event.source);       
+    }).catch(errorHandlerFunction);
+}
+```
+
+## Remove an event handler
+
+The following code sample registers an event handler for the `onParagraphChanged` event in the document and defines the `handleChange` function that will run when the event occurs. It also defines the `deregisterEventHandler()` function that can subsequently be called to remove that event handler. Note that the `RequestContext` used to create the event handler is needed to remove it.
+
+```js
+let eventContext;
+
+async function registerEventHandler() {
+  await Word.run(async (context) => {
+    eventContext = context.document.onParagraphChanged.add(handleChange);
+
+    await context.sync();
+    console.log("Event handler successfully registered for onParagraphChanged event in the document.");
+  });
+}
+
+async function handleChange(event: Word.ParagraphChangedEventArgs) {
+  await Word.run(async (context) => {
+    await context.sync();
+    console.log(`${event.type} event was detected.`);
+  });
+}
+
+async function deregisterEventHandler() {
+  await Word.run(eventContext.context, async (context) => {
```

Comment:
Consider explicitly calling out that the add-in needs to use the same context object on which the handler was registered. This surprises a lot of people

Reply:
Added comment accordingly